### PR TITLE
Guest auth frontend for omniport

### PR DESCRIPTION
**Pull request by rish9101**

### CODE REVIEW

File: src/actions/index.js [hunk 1]
```diff
@@ -33,13 +34,41 @@ export const userLogout = (_, callback) => {
...
+
+      for (var index = 0; index < roles.length; ++index) {
+        var role = roles[index];
+        if(role.role == "Guest"){
+          isGuest = true;
+          break;
+        }
+      }
+      
+      console.log(isGuest)
```
Comment (pradumangoyal) [inline src/actions/index.js]: Remove consoles

File: src/actions/index.js [hunk 2]
```diff
@@ -33,13 +34,41 @@ export const userLogout = (_, callback) => {
   }
 }
 
+export const guestUserLogin = (data, callback) => {
+  return async dispatch => {
```
Comment (pradumangoyal) [inline src/actions/index.js:40]: Implement this thing using JS promises

File: src/actions/index.js [hunk 3]
```diff
@@ -33,13 +34,41 @@ export const userLogout = (_, callback) => {
...
+}
+
 export const whoami = () => {
   return async (dispatch, getState) => {
     try {
       const res = await axios.get(getWhoAmIApi())
+
+      const roles = res.data.roles
+      
+      var isGuest =false;
```
Comment (pradumangoyal) [inline src/actions/index.js]: Use let

File: src/actions/index.js [hunk 4]
```diff
@@ -33,13 +34,41 @@ export const userLogout = (_, callback) => {
...
+
+      for (var index = 0; index < roles.length; ++index) {
```
Comment (pradumangoyal) [inline src/actions/index.js]: You should use an already written function in the `formula_one` for the same purpose: https://omniport.readthedocs.io/en/latest/references/formula_one/functions/if_role.html#if-role

**Review by pradumangoyal — CHANGES_REQUESTED** [review]
Please push the required changes at the earliest

**Comment by pradumangoyal** [comment]
This can be merged according to frontend coding practices. @ayushjn20 Review this and merge it with the PR on the backend asap.

File: src/components/login.js [hunk 5]
```diff
@@ -74,9 +74,25 @@ export class Login extends Component {
     }
   }
 
-  render () {
-    const { username, password, type, focus, error, loading, illustrationStyle } = this.state
+  guestSubmit = () => {
+    const url = this.state
+    console.log(url)
```
Comment (liquidslr) [inline src/components/login.js]: Remove all the debug statements before pushing the code

File: src/actions/index.js [hunk 6]
```diff
@@ -33,13 +36,34 @@ export const userLogout = (_, callback) => {
...
+
+      if (ifRole(roles, 'Guest') !== 'NOT_ROLE') {
+        isGuest = true
+      }
+
       dispatch({
         type: 'LOG_IN',
-        payload: res.data
+        payload: res.data,
+        isGuestAuth: isGuest
```
Comment (liquidslr) [inline src/actions/index.js:63]: can be shortened by using a ternary operator

File: src/components/login.js [hunk 7]
```diff
@@ -74,9 +74,25 @@ export class Login extends Component {
     }
   }
 
-  render () {
-    const { username, password, type, focus, error, loading, illustrationStyle } = this.state
+  guestSubmit = () => {
+    const url = this.state
```
Comment (liquidslr) [inline src/components/login.js]: Use destructuring

**Review by liquidslr — CHANGES_REQUESTED** [review]

**Comment by pradumangoyal** [comment]
Value of the new prop should be `False` by default so that it doesn't change the expected behaviour of the already present apps.

**Review by pradumangoyal — APPROVED** [review]

**Review by liquidslr — APPROVED** [review]

**Pull request merged**